### PR TITLE
towards 100% test coverage

### PR DESCRIPTION
**Pull request by dariorussi**
## Description 

Some cleanup and 100% test coverage for `message.move`

## Test plan 

This is all about testing...

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-framework/packages/bridge/sources/message.move [hunk 1]
```diff
@@ -246,9 +242,6 @@ module bridge::message {
         token_type: u8,
         amount: u64
     ): BridgeMessage {
-        // TODO: add test case for invalid chain id
-        // TODO: add test case for invalid chain id
-        // TODO: replace with `chain_ids::is_valid_chain_id()`
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/message.move:251]: do we have a test case for this?

File: crates/sui-framework/packages/bridge/sources/message.move [hunk 2]
```diff
@@ -287,8 +280,6 @@ module bridge::message {
         seq_num: u64,
         op_type: u8,
     ): BridgeMessage {
-        // TODO: add test case for invalid chain id
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/message.move:290]: ditto

File: crates/sui-framework/packages/bridge/sources/message.move [hunk 3]
```diff
@@ -315,8 +306,6 @@ module bridge::message {
         blocklist_type: u8,
         validator_ecdsa_addresses: vector<vector<u8>>,
     ): BridgeMessage {
-        // TODO: add test case for invalid chain id
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/message.move:318]: ditto

File: crates/sui-framework/packages/bridge/sources/message.move [hunk 4]
```diff
@@ -353,9 +342,6 @@ module bridge::message {
         sending_chain: u8,
         new_limit: u64,
     ): BridgeMessage {
-        // TODO: add test case for invalid chain id
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/message.move:356]: ditto

File: crates/sui-framework/packages/bridge/sources/message.move [hunk 5]
```diff
@@ -384,8 +370,6 @@ module bridge::message {
         seq_num: u64,
         new_price: u64,
     ): BridgeMessage {
-        // TODO: add test case for invalid chain id
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/message.move:387]: ditto

File: crates/sui-framework/packages/bridge/tests/message_tests.move [hunk 6]
```diff
@@ -243,12 +241,10 @@ module bridge::message_tests {
         assert!(
             bridge_limit.update_bridge_limit_payload_receiving_chain()
                 == chain_ids::sui_testnet(),
-            0,
         );
         assert!(
             bridge_limit.update_bridge_limit_payload_sending_chain()
                 == chain_ids::eth_sepolia(),
-            0,
```
[the remark this answers is not in the document]
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/tests/message_tests.move:251]: move language changed to smart error which will report the line number and module where the error happened and as such we do not need the error code any longer which only makes things confusing

File: crates/sui-framework/packages/bridge/tests/message_tests.move [hunk 7]
```diff
@@ -453,10 +464,276 @@ module bridge::message_tests {
...
+
+    #[test]
+    #[expected_failure(abort_code = bridge::message::EInvalidPayloadLength)]
+    fun test_bad_token_bridge() {
+        let sender_address = address::from_u256(100);
+        let mut scenario = test_scenario::begin(sender_address);
+        let ctx = test_scenario::ctx(&mut scenario);
+        let coin = coin::mint_for_testing<USDC>(12345, ctx);
+        let mut sender = address::to_bytes(sender_address);
+        sender.append(address::to_bytes(sender_address));
```
[the remark this answers is not in the document]
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/tests/message_tests.move:611]: I'll change the test function name which is actually pretty lame now

File: crates/sui-framework/packages/bridge/sources/message.move (hunk 1 above)
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/message.move:251]: added

File: crates/sui-framework/packages/bridge/sources/message.move (hunk 2 above)
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/message.move:290]: added

File: crates/sui-framework/packages/bridge/sources/message.move (hunk 4 above)
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/message.move:356]: added

File: crates/sui-framework/packages/bridge/sources/message.move (hunk 5 above)
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/message.move:387]: added

**Comment by dariorussi** [comment]
added check for invalid chain except for the `extract` methods which are more of an invariant and need a bit more thoughts. Basically I do not know if we can add tests for those

File: crates/sui-framework/packages/bridge/tests/message_tests.move (hunk 7 above)
[the remark this answers is not in the document]
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/tests/message_tests.move]: also added a comment

File: crates/sui-framework/packages/bridge/tests/message_tests.move (hunk 6 above)
[the remark this answers is not in the document]
Comment (longbowlu) [reply crates/sui-framework/packages/bridge/tests/message_tests.move:251]: oh wow!

**Comment by longbowlu** [comment]
> added check for invalid chain except for the `extract` methods which are more of an invariant and need a bit more thoughts. Basically I do not know if we can add tests for those

I think a couple of regression tests are sufficient for those

**Review by longbowlu — APPROVED** [review]
thanks!

**Comment by dariorussi** [comment]
> > added check for invalid chain except for the `extract` methods which are more of an invariant and need a bit more thoughts. Basically I do not know if we can add tests for those
> 
> I think a couple of regression tests are sufficient for those

there is actually a discussion about adding signing api for testing in move, that would solve this issue and be a great story, we'll see. 
Otherwise I agree with adding few cases and we have talked to @patrickkuo about it as well

**Pull request merged**